apps/admin-portal/src/app/resources/overview/resource-types/recipes/recipe.form.ts

```diff
@@ -47,48 +58,73 @@ export class RecipeForm extends BaseResourceFormComponent {
     protected override getResource(resourceId: string) {
         if (resourceId === TEMP_RESOURCE_ID) return of(TEMP_RECIPE);
         this.isLoading.set(true);
-        return this.recipesService.getById(resourceId);
+        return this.recipesService.getById(resourceId).pipe(map((recipe) => recipe.toDto()));
     }
 
     protected override populateForm() {
-        const { craftingTime, technologyTree, technologyPoints } = this.resourceService.resource() as Recipe;
+        const { craftingTime, technologyTreeId, technologyPoints, inputs, outputs } =
+            this.resourceService.resource() as RecipeDto;
 
         this.form.reset({
             craftingTime: craftingTime,
-            technologyTreeId: technologyTree?.id ?? '',
+            technologyTreeId: technologyTreeId ?? '',
             technologyPoints: technologyPoints,
         });
+
+        this.form.controls.inputs.clear();
+        (inputs ?? []).forEach((dto, index) => {
+            this.form.controls.inputs.insert(index, addRecipeItem(this.environmentInjector, 'input', dto));
+        });
+
+        this.form.controls.outputs.clear();
+        (outputs ?? []).forEach((dto, index) => {
+            this.form.controls.outputs.insert(index, addRecipeItem(this.environmentInjector, 'output', dto));
+        });
```

Comment:
Providing the initial form value through the reset function didn't seem to work for setting the in- and output values properly, but this did